src/http_server.cc

```diff
@@ -1453,13 +1467,8 @@ HTTPAPIServer::HandleRepositoryControl(
     }
   }
 
-  if (err == nullptr) {
-    evhtp_send_reply(req, EVHTP_RES_OK);
-  } else {
-    EVBufferAddErrorJson(req->buffer_out, err);
-    evhtp_send_reply(req, EVHTP_RES_BADREQ);
-    TRITONSERVER_ErrorDelete(err);
-  }
+  HTTP_RESPOND_IF_ERR(req, err);
```

Comment:
is there a missing if / else here - to avoid sending OK if we already responded with error?

Reply:
The `_IF_ERR` variant checks `err != nullptr`, and returns early if there was an error, so the OK won't get hit.

Reply:
I'll fix a couple of the spots where its inside a redundant if block

Reply:
Maybe we should rename to 'RETURN_AND_RESPOND_IF_ERR' and 'RETURN_AND_RESPONE_WITH_ERR' 

to make consistent with other RETURN_IF_ERR macros?

Reply:
I don't mind either way. Can rename to RETURN prefix if you think that's more clear.

Reply:
Renamed